pkg/cloudprovider/drift.go

```diff
@@ -40,11 +40,57 @@ import (
 )
 
 const (
+	NodeClassDrift    cloudprovider.DriftReason = "NodeClassDrift"
```

Comment:
While you're at it, and adding a new one, do you want to add a prefix to these?

Reply:
Could use the same Group you are using elsewhere?
https://github.com/Azure/karpenter-provider-azure/blob/07052f3cbdcd6944ee5d9dde8e9b7b089aa3ffec/pkg/apis/apis.go#L31

Reply:
I do like the idea of prefix, but don't want to do it in a rush (e.g. would be good to check if there are limits on length somewhere, etc.); let's circle back to this later